### PR TITLE
merge search reqs v1

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>

### CODE REVIEW

File: internal/querynode/query_collection.go [hunk 1]
```diff
@@ -57,6 +57,21 @@ type queryMsg interface {
 	RecordSpan() time.Duration
 }
 
+type MergedReqMsg struct {
```
Comment (xiaocai2333) [inline internal/querynode/query_collection.go:60]: delete it, use by czs.

File: internal/querynode/query_collection.go [hunk 2]
```diff
@@ -543,176 +585,64 @@ func (q *queryCollection) receiveQueryMsg(msg queryMsg) error {
...
+// TODO @xiaocai2333: implement it.
+func (q *queryCollection) mergeSearchReq(mergedMsg queryMsg, msg queryMsg) queryMsg {
+	return msg
 }
 
 func (q *queryCollection) doUnsolvedQueryMsg() {
 	log.Debug("starting doUnsolvedMsg...", zap.Any("collectionID", q.collectionID))
+	emptySearchQueue := true
 	for {
+		go func() {
```
Comment (xiaocai2333) [inline internal/querynode/query_collection.go:605]: move it outside the for loop.

File: internal/querynode/query_collection.go [hunk 3]
```diff
@@ -543,176 +570,66 @@ func (q *queryCollection) receiveQueryMsg(msg queryMsg) error {
...
-	return nil
+// TODO @xiaocai2333: implement it.
+func (q *queryCollection) mergeSearchReq(mergedMsg queryMsg, msg queryMsg) queryMsg {
+	return msg
 }
 
 func (q *queryCollection) doUnsolvedQueryMsg() {
 	log.Debug("starting doUnsolvedMsg...", zap.Any("collectionID", q.collectionID))
+	emptySearchQueue := true
+	go func() {
```
Comment (czs007) [inline internal/querynode/query_collection.go:589]: what 's  the purpose of this goroutine?

File: internal/querynode/query_collection.go [hunk 4]
```diff
@@ -398,6 +422,34 @@ func (q *queryCollection) consumeQuery() {
 	}
 }
 
+func (q *queryCollection) doSearchRequests() {
+	for {
+		select {
```
Comment (czs007) [inline internal/querynode/query_collection.go:427]: how to exit  normaly ?

Comment (xiaocai2333) [reply internal/querynode/query_collection.go:433]: sorry, my bad, I forget it. Now it has been fixed.

File: internal/querynode/query_collection.go (hunk 3 above)
Comment (xiaocai2333) [reply internal/querynode/query_collection.go:598]: When there is no task to perform for the goroutine for searching `doSearchRequests`, I hope that a search request can be sent quickly. The purpose of this goroutine is to make the goroutine ` doUnsolvedQueryMsg` know  that the goroutine `doSearchRequests` wants a search requests.

File: internal/querynode/query_collection.go [hunk 5]
```diff
@@ -449,155 +509,102 @@ func (q *queryCollection) adjustByChangeInfo(msg *msgstream.SealedSegmentsChange
...
-			zap.Int64(log.BenchmarkCollectionID, collectionID),
+			zap.Int64(log.BenchmarkCollectionID, q.collectionID),
 			zap.Int64(log.BenchmarkMsgID, msg.ID()), zap.Int64(log.BenchmarkDuration, time.Now().UnixNano()))
 
 	default:
-		err := fmt.Errorf("receive invalid msgType = %d", msgType)
+		err := fmt.Errorf("receive invalid msgType = %d", msg.Type())
 		return err
 	}
+	// why can receive the msg that does not belong to this collection?
```
Comment (bigsheeper) [inline internal/querynode/query_collection.go:536]: delete this?

**Pull request merged**